slidechain/export.go

```diff
@@ -90,15 +154,47 @@ func (c *Custodian) pegOutFromExports(ctx context.Context) {
 			}
 
 			log.Printf("pegging out export %x: %d of %s to %s", txid, amounts[i], asset.String(), exporters[i])
-			// TODO(vniu): flag txs that fail with unretriable errors in the db
-			err = c.pegOut(ctx, exporter, asset, int64(amounts[i]), tempID, xdr.SequenceNumber(seqnums[i]))
-			if err != nil {
-				log.Fatalf("pegging out tx: %s", err)
+			var peggedOut pegOutState
+			for i := 0; i < 5; i++ {
```

Comment:
Danger: this `i` shadows the `i` in the outer loop.

With the `break`s and `continue`s in this loop, and the other logic errors I've pointed out, I think you're better off factoring the retry logic out as a separate function.

Better still, don't retry in a loop here at all. Instead, if you encounter a retriable error, flag it as such in the database and then move on to the next `exports` record. Come back around to the retriable record later via the db query, which should produce both `pegOutNotYet` and `pegOutRetry` rows.